packages/wallet-ui/src/store/wallet/wallet.ts

```diff
@@ -112,6 +107,60 @@ export const deposit = (
   }
 };
 
+export enum WalletWithdrawTransition {
+  CheckWallet = "WALLET_WITHDRAW_CHECK_WALLET",
+  WaitForFunds = "WALLET_WITHDRAW_WAITING_FOR_FUNDS"
+}
+
+export const withdraw = (
+  transaction: Deposit,
+  provider: Web3Provider,
+  history?: History
+): ThunkAction<
+  void,
+  ApplicationState,
+  null,
+  Action<ActionType | WalletWithdrawTransition>
+> => async dispatch => {
+  try {
+    // 1. Ask Metamask to do the deposit. !
+    log('withdraw', transaction)
```

Comment:
Should we keep these logs?

Reply:
I don't know, I would leave at least the second one (which has the transaction response, with the hash) for debugging purposes, until we have that working reliably